circleci-provision-scripts/gcloud.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+function install_gcloud() {
+        # Gcloud installation script needs to instal .config/gcloud dir under user directory
+        (cat <<'EOF'
+sudo bash -c 'curl "https://sdk.cloud.google.com" | CLOUDSDK_CORE_DISABLE_PROMPTS=1 CLOUDSDK_INSTALL_DIR=/opt bash'
+EOF
+	) | as_user bash
```

Comment:
~~ The combination of `sudo` inside the heredoc with `as_user` (i.e. as ubuntu) outside and multiple `bash` invocations is a bit confusing.  Are they all needed?  How is it different from

```bash
curl "https://sdk.cloud.google.com" | CLOUDSDK_CORE_DISABLE_PROMPTS=1 CLOUDSDK_INSTALL_DIR=/opt bash
```